app/code/Magento/MediaGalleryUi/Model/Directories/GetFolderTree.php

```diff
@@ -84,8 +86,11 @@ private function getDirectories(): array
             }
 
             $pathArray = explode('/', $path);
+            $displayName = strlen(end($pathArray)) > self::NAME_LENGTH ?
+                         substr(end($pathArray), 0, self::NAME_LENGTH) . "..." :
+                         end($pathArray);
             $directories[] = [
-                'data' => count($pathArray) > 0 ? end($pathArray) : $path,
+                'data' => count($pathArray) > 0 ? $displayName : $path,
```

Comment:
Not sure it's a right way to trim name on the backend, it would be better adapt frontend application to handle long names. Let's imagine that someone will decide to extend frontend implementation and provide new design (where long names will look good) and with such approach, we are limiting 3d party developers to use trimmed names only.

Reply:
agree but there is some limitation in a frontend so i ecided it can be more esier to fix thid in a backend part, but i will try one more time fix this behavior on a frontend, by the way newest version of jstree already have this functionality :) but magento still using old version of jstree :(

Reply:
https://github.com/vakata/jstree/commit/3c5bf416d85a32392c3b330ad20e386357fdc367

Reply:
@sidolov i'v changed implementation, instead of cutting folders as it was done in old media gallery and there is also this issue here ->
![Screenshot_2020-08-20 Default Category (ID 2) Categories Inventory Catalog Magento Admin](https://user-images.githubusercontent.com/2481206/90763726-dfeaf580-e2ef-11ea-9078-c7d00c452c2e.png)

so when we have a long folders and the block is too small for it i had added ability to scroll, as we can have alot of folders created one in one and soon or later we also have same issue, so i think only adding ability to vertical scrolling when there is a lot folder will be correct

Reply:
@Nazar65 sounds good to me! Look like it's the only solution in such case.